src/py/_pyodide/console.py

```diff
@@ -97,8 +98,13 @@ def __init__(
 
     def __call__(self, source, filename, symbol) -> CodeRunner:  # type: ignore
         return_mode = self.return_mode
-        if self.quiet_trailing_semicolon and should_quiet(source):
-            return_mode = None
+        try:
+            if self.quiet_trailing_semicolon and should_quiet(source):
+                return_mode = None
+        except TokenError:
+            # Invalid code, let the Python parser throw the error later.
+            pass
```

Comment:
Is this related to this PR?

Reply:
Nope, just saw this issue in the process of working on this PR. I will move it into a separate PR.